src/modelfree/shaping_wrappers.py

```diff
@@ -7,66 +7,52 @@
+    """
+    A more direct interface for shaping the reward of the attacking agent.
+    - shaping_params schema: {'sparse': {k: v}, 'dense': {k: v}, **kwargs}
+    """
     def __init__(self, venv, agent_idx, shaping_params, batch_size, reward_annealer=None):
         super().__init__(venv)
-        # structure: {'sparse': {k: v}, 'dense': {k: v}, **kwargs}
         self.shaping_params = shaping_params
         self.reward_annealer = reward_annealer
-        self.counter = 0
         self.batch_size = batch_size
         self.agent_idx = agent_idx
 
         self.ep_rew_dict = defaultdict(list)
         self.step_rew_dict = defaultdict(lambda: [[] for _ in range(self.num_envs)])
 
-    @classmethod
-    def get_shaping_params(cls, shaping_params, env_name):
-        path_stem = 'experiments/rew_configs/'
-        default_configs = {
-            'multicomp/SumoHumans-v0': 'default_hsumo.json',
-            'multicomp/SumoHumansAutoContact-v0': 'default_hsumo.json'
-        }
-        assert env_name in default_configs, \
-            "No default reward shaping config found for env_name:{}".format(env_name)
-        fname = os.path.join(path_stem, default_configs[env_name])
-        final_shaping_params = json.load(open(fname))
-
-        if shaping_params is not None:
-            config = json.load(open(shaping_params))
-            final_shaping_params['sparse'].update(config['sparse'])
-            final_shaping_params['dense'].update(config['dense'])
-            final_shaping_params['rew_shape_anneal_frac'] = config.get('rew_shape_anneal_frac')
-        return final_shaping_params
-
-    def _log_sparse_dense_rewards(self):
-        if self.counter == self.batch_size * self.num_envs:
-            num_episodes = len(self.ep_rew_dict['sparse']['reward_remaining'])
-            dense_terms = self.shaping_params['dense'].keys()
-            for term in dense_terms:
-                assert len(self.ep_rew_dict[term]) == num_episodes
-            ep_dense_mean = sum([sum(self.ep_rew_dict[t]) for t in dense_terms]) / num_episodes
-            logger.logkv('epdensemean', ep_dense_mean)
-
-            sparse_terms = self.shaping_params['sparse'].keys()
-            for term in sparse_terms:
-                assert len(self.ep_rew_dict[term]) == num_episodes
-            ep_sparse_mean = sum([sum(self.ep_rew_dict[t]) for t in sparse_terms]) / num_episodes
-            logger.logkv('epsparsemean', ep_sparse_mean)
-
-            c = self.reward_annealer()
-            logger.logkv('rew_anneal', c)
-            ep_rew_mean = c * ep_dense_mean + (1 - c) * ep_sparse_mean
-            logger.logkv('eprewmean_true', ep_rew_mean)
-
-            for rew_type in self.ep_rew_dict:
-                self.ep_rew_dict[rew_type] = []
-            self.counter = 0
+    def log_sparse_dense_rewards(self, locals, globals):
+        """Logs various metrics. This is given as a callback to PPO2.learn()"""
+        _ = [locals, globals]
```

Comment:
You could also have this not take `locals, globals` and just have a `lambda` function swallow those arguments when you add it to the callback: `lambda locals, globals: single_env.log_sparse_dense_rewards()`. This seems cleaner unless it might need to know about `locals` or `globals` in the future.